utils/build/docker/nodejs/parametric/server.js

```diff
@@ -35,96 +35,68 @@ const otelStatusCodes = {
   for (const [key, value] of Object.entries(http_headersDict)) {
-      http_headers.push([key, value])
+      http_headers.push([key, value]);
   }
 
   res.json({ http_headers });
 });
 
-// Additional Endpoints
-app.post('/trace/span/start', (req, res) => {
+app.post('/trace/span/extract_headers', (req, res) => {
   const request = req.body;
-  let parent
-
-  if (request.parent_id) parent = spans[request.parent_id]
-
-  if (request.origin) {
-      const traceId = parent?.traceId
-      const parentId = parent?.parentId
+  const http_headers = request.http_headers || [];
+  // Node.js HTTP headers are automatically lower-cased, simulate that here.
+  const linkHeaders = Object.fromEntries(http_headers.map(([k, v]) => [k.toLowerCase(), v]));
+  const extracted = tracer.extract('http_headers', linkHeaders);
 
-      parent = new SpanContext({
-          traceId,
-          parentId
-      })
-      parent.origin = request.origin
+  let extractedSpanID = null;
+  if (extracted && extracted._spanId) {
+    extractedSpanID = extracted.toSpanId();
+    ddContext[extractedSpanID] = extracted;
   }
 
-  const http_headers = request.http_headers || []
-  // Node.js HTTP headers are automatically lower-cased, simulate that here.
-  const convertedHeaders = {}
-  for (const [key, value] of http_headers) {
-      convertedHeaders[key.toLowerCase()] = value
-  }
+  res.json({ span_id: extractedSpanID });
+});
 
-  const extracted = tracer.extract('http_headers', convertedHeaders)
-  let extractedLinks = []
-  if (extracted !== null) {
-    parent = extracted
-    extractedLinks = parent._links
-  }
+app.post('/trace/span/start', (req, res) => {
+  const request = req.body;
+  let parent = spans[request.parent_id] || ddContext[request.parent_id];
 
-  const tags = { service: request.service }
+  const tags = {service: request.service, resource: request.resource};
   for (const [key, value] of request.span_tags) tags[key] = value
 
   const span = tracer.startSpan(request.name, {
     type: request.type,
-    resource: request.resource,
     childOf: parent,
     tags
-  })
-
-  for (const link of extractedLinks || []) {
-    span.addLink(link.context, link.attributes)
-  }
+  });
 
-  for (const link of request.links || []) {
-    const linkParentId = link.parent_id;
-    if (linkParentId) {
-      const linkParentSpan = spans[linkParentId];
-      span.addLink(linkParentSpan.context(), link.attributes);
-    } else {
-      const linkHeaders = link.http_headers || {};
-      const convertedLinkHeaders = {}
-      for (const [key, value] of linkHeaders) {
-        convertedLinkHeaders[key.toLowerCase()] = value
-      }
-      const linkExtracted = tracer.extract('http_headers', convertedLinkHeaders);
-      if (linkExtracted) {
-        span.addLink(linkExtracted, link.attributes);
-      }
-    }
+  if (ddContext[request.parent_id]) {
+    for (const link of ddContext[request.parent_id]._links || []) span.addLink(link.context, link.attributes)
```

Comment:
Currently on the master branch, spanContexts don't store ._links. Links are stored on the span. I know Matt is currently working on a PR to allow for storing of links on span contexts but as of now this will always return an empty array